src/crypto/PSASessionKeystore.cpp

```diff
@@ -192,5 +189,72 @@ void PSASessionKeystore::DestroyKey(HkdfKeyHandle & key)
     keyId = PSA_KEY_ID_NULL;
 }
 
+#if CHIP_CONFIG_ENABLE_ICD_CIP
+CHIP_ERROR PSASessionKeystore::PersistICDKey(Aes128KeyHandle & key)
+{
+    CHIP_ERROR err;
+    AesKeyAttributes attrs;
+    psa_key_id_t previousKeyId = key.As<psa_key_id_t>();
+    psa_key_attributes_t previousKeyAttrs;
+
+    psa_get_key_attributes(previousKeyId, &previousKeyAttrs);
+    // Exit early if key is already persistent
+    if (psa_get_key_lifetime(&previousKeyAttrs) == PSA_KEY_LIFETIME_PERSISTENT)
+    {
+        ExitNow(err = CHIP_NO_ERROR);
+    }
+
+    SuccessOrExit(err = Crypto::FindFreeKeySlotInRange(key.AsMutable<psa_key_id_t>(), to_underlying(KeyIdBase::ICDAesKeyRangeStart),
+                                                       kMaxICDClientKeys));
+
+    SuccessOrExit(err = attrs.SetKeyPersistence(key.As<psa_key_id_t>()));
+    VerifyOrExit(psa_copy_key(previousKeyId, &attrs.Get(), &key.AsMutable<psa_key_id_t>()) == PSA_SUCCESS,
+                 err = CHIP_ERROR_INTERNAL);
+
+    psa_destroy_key(previousKeyId);
+
+exit:
+    if (err != CHIP_NO_ERROR)
+    {
+        psa_destroy_key(previousKeyId);
+        psa_destroy_key(key.As<psa_key_id_t>());
```

Comment:
This seems unnecessary as in the case of error the key doesn't have a chance to be created.

Reply:
changed the approach on PersistICDKey. Destroying of keys should be fine now as well